iib/web/models.py

```diff
@@ -1054,6 +1076,97 @@ def get_mutable_keys(self):
         return rv
 
 
+class RequestMergeIndexImage(Request, RequestIndexImageMixin):
+    """An "merge-index-image" build request."""
+
+    __tablename__ = 'request_merge_index_image'
+
+    id = db.Column(db.Integer, db.ForeignKey('request.id'), autoincrement=False, primary_key=True)
+    source_from_index_id = db.Column(db.Integer, db.ForeignKey('image.id'), nullable=True)
+    source_from_index = db.relationship('Image', foreign_keys=[source_from_index_id], uselist=False)
+    target_index_id = db.Column(db.Integer, db.ForeignKey('image.id'), nullable=True)
+    target_index = db.relationship('Image', foreign_keys=[target_index_id], uselist=False)
+    deprecation_list = db.relationship('Image', secondary=BundleDeprecation.__table__)
+
+    __mapper_args__ = {
+        'polymorphic_identity': RequestTypeMapping.__members__['merge_index_image'].value,
+    }
+
+    @classmethod
+    def from_json(cls, kwargs, batch=None):
+        """
+        Handle JSON requests for the merge-index-image API endpoint.
+
+        :param dict kwargs: the JSON payload of the request.
+        :param Batch batch: the batch to specify with the request.
+        """
+        request_kwargs = deepcopy(kwargs)
+
+        bundles = request_kwargs.pop('deprecation_list', [])
```

Comment:
Let's use `deprecation_list` as the variable name instead of bundles to avoid confusion.

Reply:
Fixed